sqlmesh/core/state_sync/common.py

```diff
@@ -215,25 +219,13 @@ def _ensure_no_gaps(
             and target_snapshots_by_name[s.name].version != s.version
         }
 
-        changed_version_target_snapshots = [
-            t for t in target_snapshots if t.name in changed_version_prev_snapshots_by_name
-        ]
-
-        all_snapshot_intervals = self.get_snapshot_intervals(
-            [*changed_version_prev_snapshots_by_name.values(), *changed_version_target_snapshots],
-        )
-
-        hydrated_prev_snapshots = Snapshot.hydrate_with_intervals_by_version(
-            self.get_snapshots(changed_version_prev_snapshots_by_name.values()).values(),
-            all_snapshot_intervals,
-        )
-        hydrated_target_snapshots = Snapshot.hydrate_with_intervals_by_version(
-            changed_version_target_snapshots, all_snapshot_intervals
-        )
-        hydrated_target_snapshots_by_name = {s.name: s for s in hydrated_target_snapshots}
+        prev_snapshots = self.get_snapshots(
+            changed_version_prev_snapshots_by_name.values(),
+            refresh=True,
```

Comment:
Can we just call `self._get_snapshots()` here instead of `refresh=True` and bypass cache altogether?

Reply:
Perhaps do the same here: https://github.com/TobikoData/sqlmesh/blob/1584aa5c5d1a15a2dec80eb8ade1e61188f13ab6/sqlmesh/core/state_sync/common.py#L93